.github/workflows/starknet-js-tests.yml

```diff
@@ -0,0 +1,37 @@
+name: Run Tests
+
+on:
+  workflow_dispatch:
+  workflow_call:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    env:
+      BINARY_PATH: ../target/release/madara
+    steps:
+      - uses: actions/checkout@v3
+      - uses: Swatinem/rust-cache@v2
+        with:
+          shared-key: "cache"
+          save-if: false
+      - uses: actions/cache@v3
+        with:
+          path: target/release/madara
+          key:
+            ${{ runner.os }}-cargo-${{ hashFiles('Cargo.lock') }}-${{
+            github.run_id }}
+          fail-on-cache-miss: true
+      - name: Setup dev chain
+        run: |
+          ./target/release/madara setup --chain=dev --from-remote
+      - name: Run starknet-js test
+        run: |-
+          ./target/release/madara --dev --tmp --rpc-external --execution native --pool-limit=100000 --pool-kbytes=500000 --rpc-methods=unsafe --rpc-cors=all --in-peers=0 --out-peers=1 --no-telemetry &
```

Comment:
if u run `--dev` u don't need to specify the 
- `--tmp`
- `--rpc-external`
- `--rpc-methods`
- `--rpc-cors`

why are u limiting the pool and the number of peers? and why are u specifying `--no-telemetry`?

Reply:
```suggestion
          ./target/release/madara --dev --execution native &
```

Reply:
The blocks weren't sealing, so I eventually copied the contents from the run script. Also didn't want to burn through the workflow hours excessively